### PR TITLE
GH-45047: [CI][Python][Packaging] Test 3.12 wheels on Ubuntu 24.04

### PR DESCRIPTION
**Pull request by pitrou**
### Rationale for this change

We already test some Python wheels on Ubuntu 20.04 or 22.04, depending on the system Python version. Also do so with Ubuntu 24.04.

### Are these changes tested?

Yes, by definition.

### Are there any user-facing changes?

No.
* GitHub Issue: #45047

### CODE REVIEW

**Comment by pitrou** [comment]
@github-actions crossbow submit wheel-manylinux*cp312*

**Comment by pitrou** [comment]
@github-actions crossbow submit wheel-manylinuxcp312

**Comment by pitrou** [comment]
@github-actions crossbow submit wheel-manylinux*cp312*

**Comment by pitrou** [comment]
@github-actions crossbow submit wheel-manylinux*cp312*

**Comment by pitrou** [comment]
The `test_orc.py` segfault is due to https://github.com/apache/arrow/issues/40633 . It helped me file this fix: https://github.com/apache/arrow/pull/45051

**Comment by pitrou** [comment]
@github-actions crossbow submit wheel-manylinux-2014*

**Comment by pitrou** [comment]
@github-actions crossbow submit wheel-manylinux-2-28*

**Comment by pitrou** [comment]
@kou Would you like to take a look at this PR?

File: dev/release/setup-ubuntu.sh [hunk 1]
```diff
@@ -22,27 +22,20 @@
...
-case ${codename} in
-  focal)
-    ;;
-  *)
-    apt-get update -y -q
-    apt-get install -y -q --no-install-recommends \
-      libxsimd-dev
-    ;;
-esac
+if [ ${version} \> "20.04" ]; then
```
Comment (kou) [inline dev/release/setup-ubuntu.sh:29]: Oh, I didn't know this feature.

**Review by kou — APPROVED** [review]
+1

**Pull request merged**